pkg/k8s/mounterBuilder_test.go

```diff
@@ -0,0 +1,269 @@
+				},
+			},
+		}, nil
+	})
+	k8sMock.AddReactor("get", "deployments", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1beta1.Deployment{
+			Spec: v1beta1.DeploymentSpec{
+				Template: v1.PodTemplateSpec{
+					Spec: v1.PodSpec{
+						Volumes: []v1.Volume{
+							{
+								Name: "test-secret",
+							},
+						},
+						Containers: []v1.Container{
+							{
+								Name: "test-service",
+								VolumeMounts: []v1.VolumeMount{
+									{
+										Name: "test-secret",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}, nil
+	})
+	k8sMock.AddReactor("get", "secrets", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1.Secret{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: "test-secret",
+			},
+			Data: map[string][]byte{},
+		}, nil
+	})
+	k8sMock.AddReactor("Update", "deployments", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1beta1.Deployment{
+			Spec: v1beta1.DeploymentSpec{
+				Template: v1.PodTemplateSpec{
+					Spec: v1.PodSpec{
+						Volumes: []v1.Volume{},
+						Containers: []v1.Container{
+							{
+								Name:         "fh-sync-server",
+								VolumeMounts: []v1.VolumeMount{},
+							},
+						},
+					},
+				},
+			},
+		}, nil
+	})
+	return k8sMock
+}
+
+func TestMount(t *testing.T) {
+	cases := []struct {
+		Name      string
+		K8sClient func() *fake.Clientset
+		Namespace string
+		Service   string
+		Secret    string
+		Validate  func(t *testing.T, mountRes error)
+	}{
+		{
+			Name:      "Valid mount request",
+			K8sClient: mockUnmountedK8sClient,
+			Namespace: "test-namespace",
+			Service:   "test-service",
+			Secret:    "test-secret",
+			Validate: func(t *testing.T, mountRes error) {
+				if mountRes != nil {
+					t.Fatalf("Did not expect error mounting in testCase: Valid mount request, got: %v", mountRes)
+				}
+			},
+		},
+		{
+			Name:      "Bad service name to mount request",
+			K8sClient: mockUnmountedK8sClient,
+			Namespace: "test-namespace",
+			Service:   "test-bad-service",
+			Secret:    "test-secret",
+			Validate: func(t *testing.T, mountRes error) {
+				if mountRes == nil {
+					t.Fatalf("expected error when providing bad clientService name, but got none")
+				}
+				if mountRes.Error() != "k8s.mm.Mount -> could not find container in deployment with name: test-bad-service" {
+					t.Fatalf("Expected 'k8s.mm.Mount -> could not find container in deployment with name: test-bad-service' but got: %s", mountRes.Error())
+				}
+			},
+		},
+		{
+			Name:      "Bad secret name to mount request",
+			K8sClient: mockUnmountedK8sClient,
+			Namespace: "test-namespace",
+			Service:   "test-service",
+			Secret:    "test-bad-secret",
+			Validate: func(t *testing.T, mountRes error) {
+				if mountRes == nil {
+					t.Fatalf("expected error when providing bad secret name, but got none")
+				}
+				if mountRes.Error() != "k8s.mm.Mount -> could not find secret: test-bad-secret" {
```

Comment:
I am not personally the biggest fine of testing an error message as these can make the tests pretty fragile. I would go for simply testing that an error was returned when expected. WDYT

Reply:
Sure, I was just concerned about a false positive if a different error is returned, but I'm happy either way.